Code/Framework/AzCore/AzCore/Math/IntersectSegment.cpp

```diff
@@ -923,6 +923,81 @@ int AZ::Intersect::IntersectRayCone(
     }
 }
 
+// This intersection algorithm is adapted from 'Capped Cone' by Inigo Quilez
+// ref: https://www.iquilezles.org/www/articles/intersectors/intersectors.htm and https://www.shadertoy.com/view/llcfRf
+// All algorithms/code snippets are kindly made available under the MIT License
+// https://www.iquilezles.org/www/index.htm - 'all code snippets you'll find are under the MIT license so you can easily reuse them.'
+int AZ::Intersect::IntersectRayCone2(
+    const Vector3& rayOrigin, const Vector3& rayDir, const Vector3& coneApex, const Vector3& coneDir, float coneHeight,
+    float coneBaseRadius, float& t)
+{
+    const AZ::Vector3& pa = coneApex;
+    const AZ::Vector3& pb = coneApex + coneHeight * coneDir;
+    const AZ::Vector3& ro = rayOrigin;
+    const AZ::Vector3& rd = rayDir;
+
+    float ra = 0.0f;
```

Comment:
I'd suggest either removing the math below that isn't needed because ra (radius apex) is 0, or just opening up the API to allow for the user to pass this in. I lean towards the latter because you're creating a new function anyway and it's useful functionality to have.

Reply:
I've eliminated `ra` and all expressions that were using it (please see `ManipulatorBounds.cpp` for the updated function)